src/grammar.coffee

```diff
@@ -604,7 +604,8 @@ grammar =
   ArgElisionList: [
     o 'ArgElision'
     o 'ArgElisionList , ArgElision',                                          -> $1.concat $3
-    o 'ArgElisionList OptElisions TERMINATOR ArgElision',                     -> $1.concat $2, $4
```

Comment:
Basically there was no grammar support for a `TERMINATOR` following a "leading" `Elision` (ie preceding any actual non-elision array elements)

The reason that things like this worked is because leading commas suppress `TERMINATOR`s:
```
[
  ,
  ,
  ,
]
```

So the main grammar change is to add the rule `Elision TERMINATOR` below

Then this rule needed to be split apart in order not to create a grammar conflict - basically this rule `ArgElisionList OptElisions TERMINATOR ArgElision` was covering all of these different cases:

- A typical indented array with no comma eg
```
[
  a
  b
]
```

- A typical indented array with a comma eg
```
[
  a,
  b
]
```

- A non-initial elision followed by a newline eg
```
[
  a
  ,
  ,
  b
]
```

So this just replaces the rule with a rule that only covers the first two of these cases, and then the third is handled by the addition of `Elision TERMINATOR` below